actions/setup-token/src/main.ts

```diff
@@ -0,0 +1,132 @@
+        INPUT_SLSA-PRIVATE-REPOSITORY=true \
+        INPUT_SLSA-RUNNER-LABEL="ubuntu-latest" \
+        INPUT_SLSA-BUILD-ACTION-PATH="./actions/build-artifacts-composite" \
+        INPUT_SLSA-WORKFLOW-INPUTS="{\"name1\":\"value1\",\"name2\":\"value2\",\"private-repository\":true}" \
+        nodejs ./dist/index.js
+    */
+
+    const workflowRecipient = core.getInput("slsa-workflow-recipient");
+    const privateRepository = core.getInput("slsa-private-repository");
+    const runnerLabel = core.getInput("slsa-runner-label");
+    const buildArtifactsActionPath = core.getInput("slsa-build-action-path");
+    // The workflow inputs are represented as a JSON object theselves.
+    const workflowsInputsText = core.getInput("slsa-workflow-inputs");
+
+    // Log the inputs for troubleshooting.
+    core.debug(`workflowsInputsText: ${workflowsInputsText}`);
+    core.debug(`workfowInputs: `);
+    const workflowInputs = JSON.parse(workflowsInputsText);
+    const workflowInputsMap = new Map(Object.entries(workflowInputs));
+    for (const [key, value] of workflowInputsMap) {
+      core.info(` ${key}: ${value}`);
+    }
+
+    const payload = JSON.stringify(github.context.payload, undefined, 2);
+    core.debug(`The event payload: ${payload}`);
+
+    // Construct an unsigned SLSA token.
+    const unsignedSlsaToken = {
+      version: 1,
+      context: "SLSA delegator framework",
+      builder: {
+        "private-repository": privateRepository,
+        "runner-label": runnerLabel,
+        audience: workflowRecipient,
+      },
+      github: {
+        // TODO(#1410): Re-evaluate if we need the context.
+        context: github,
+        event_name: process.env.GITHUB_EVENT_NAME,
+        run_attempt: process.env.GITHUB_RUN_ATTEMPT,
+        run_id: process.env.GITHUB_RUN_ID,
+        run_number: process.env.GITHUB_RUN_NUMBER,
+        workflow: process.env.GITHUB_WORKFLOW,
+        sha: process.env.GITHUB_SHA,
+        repository: process.env.GITHUB_REPOSITORY,
+        repository_owner: process.env.GITHUB_REPOSITORY_OWNER,
+        // TODO(#1411): Record if these become available.
+        // repository_id: process.env.GITHUB_REPOSITORY_ID,
+        // repository_owner_id: process.env.GITHUB_REPOSITORY_OWNER_ID,
+        ref: process.env.GITHUB_REF,
+        ref_type: process.env.GITHUB_REF_TYPE,
+        actor: process.env.GITHUB_ACTOR,
+      },
+      tool: {
+        actions: {
+          "build-artifacts": {
+            path: buildArtifactsActionPath,
+          },
+        },
+        inputs: workflowInputs,
+      },
+    };
+
+    // Prepare the base64 unsigned token.
+    const unsignedToken = JSON.stringify(unsignedSlsaToken, undefined);
+    const unsignedB64Token = Buffer.from(unsignedToken).toString("base64");
+    core.info(`unsignedToken: ${unsignedToken}`);
+    core.info(`unsignedB64Token: ${unsignedB64Token}`);
+
+    // Sign and prepare the base64 bundle.
+    const eventName = process.env.GITHUB_EVENT_NAME || "";
```

Comment:
@ianlewis you suggested I use `process.env["GITHUB_EVENT_NAME"]`, do you know why it's better?

Reply:
Sorry for the confusion. I actually suggested the opposite. The code here was originally `process.env["GITHUB_EVENT_NAME"]` and I suggested `process.env.GITHUB_EVENT_NAME` to be consistent with the notation used on lines 66-80. I think this is fine now.